### ISSUE TITLE
 S2372 should not trigger on properties implementing an interface and throwing NotImplementedException

### ISSUE THREAD

**Issue by jessehouwing** [SonarSource-VisualStudio/sonarlint-roslyn-analyzer#589]
With the code below:

```
        public DateTime RevisedDate
        {
            get
            {
                throw new NotImplementedException();
            }
        }
```

the rule s2372 triggers, but it probably shouldn't have, as the property is from an interface implemented (partially) by this class. It indeed breaks the contract, but at least it's obvious about it. Returning DateTime.MinValue would resolve the warning, but lead to far more unexpected behavior.

**Comment by tvsonar** [comment]
We briefly considered adding an exception for getters with only a single `throw` in them. But then decided not to implement this. We found this pattern occur in test projects, where you auto-implement an interface, but only test a part of it. 

Did you find this issue in non-test code? What's your use case? If the `RevisedDate` is not used anywhere, then why is it part of the interface?

**Comment by jessehouwing** [comment]
Indeed often in mocks and stubs. Which is also the case here. Or in places
where we need to have a getter or setter defined so that a mock can easily
override the value (we're using NSubstitute). Without the setter it's
unable to override the value, but the underlying implementation would never
actually use it, it would proxy the value from a wrapped class.

Sometimes in certain interfaces that have a lot of members where not all
are of importance to what you're trying to accomplish or where the consumer
just doesn't care. In case the consumer does start to care I prefer it to
break over accepting bogus values.

**Comment by tvsonar** [comment]
We created #590 to recognize test methods. As for the big interfaces, we'll see if we come across some common patterns that can be excluded.
